storage/mysql/admin_storage.go

```diff
@@ -472,6 +476,10 @@ func (t *adminTX) UpdateTree(ctx context.Context, treeID int64, updateFunc func(
 		return nil, err
 	}
 
+	// TODO(pavelkalinnikov): When switching TreeState to FROZEN for a
```

Comment:
@Martin2112 Does storage check `Unsequenced` table emptiness during a `DRAINING`->`FROZEN` transition, or it just trusts the caller of the API?

Reply:
No. Checking this can be prohibitively expensive on a large MySQL database. Distributed databases have other problems with this type of query as it hits all the shards.

Reply:
I assume you are talking about the counting query, like `"SELECT COUNT(*) FROM Unsequenced/SequencedLeafData"`?

However, I think that we can afford even heavy requests in the scenario when we try to switch from `DRAINING` to `FROZEN` with caution. The reason is - the tree is inactive (if called at the right moment). Checking would be just a consistency protection mechanism.

Reply:
Hm, on the other hand I don't see why a tree can't be FROZEN even if it has some extra entries in `Unsequenced/SequencedLeafData`. Probably this check should be only in the case of changing the log type.

Reply:
I think you should create a large database of say 600 million log entries and test queries like that while the log is serving requests. It's possible that an EXISTS query might work but I'd not bet on it without trying.